### PR TITLE
fix(ptp): avoid duplicate torrent artifacts

### PR DESCRIPTION
**Pull request by Audionut**
## Summary

- finalize PTP's canonical `[ptp].<release>.torrent` artifact in place instead of retaining the same-infohash `.ptp.torrent` copy
- require `announce_url` before PTP upload preparation, avoiding uploads that cannot be reconstructed for client injection
- verify source, announce/announce-list, comment, infohash, canonical path, and duplicate absence

## Testing

- `make precommit`
- `make test-go`
- `make backend`

## Summary by CodeRabbit

- **New Features**
  - PTP upload artifacts now include required tracker announce information and PTP source metadata.
- **Bug Fixes**
  - Improved PTP torrent artifact resolution during uploads.
  - Upload validation now confirms key metadata, including the prepared infohash, announce details, source, and upload URL.
  - Prevented creation of duplicate legacy torrent artifacts.
- **Tests**
  - Expanded coverage to verify PTP policy configuration and finalized upload artifacts.

### CODE REVIEW

File: internal/trackers/impl/standalone/ptp/definition_test.go
Comment (Audionut) [inline internal/trackers/impl/standalone/ptp/definition_test.go:1]: @coderabbitai Addressed "Add a focused upload-preparation test for a missing announce URL" in e090e22c1: PTP now rejects an empty announce before session or upload requests; the new stored-session regression test proves zero tracker requests. The PTP race suite passes.

File: internal/trackers/impl/standalone/ptp/definition_test.go [hunk 1]
```diff
@@ -513,11 +522,34 @@ func TestDefinitionUploadSuccess(t *testing.T) {
...
+	}
+	if registeredMeta.Comment != result.UploadedTorrents[0].TorrentURL {
+		t.Fatalf("expected registered PTP comment %q, got %q", result.UploadedTorrents[0].TorrentURL, registeredMeta.Comment)
 	}
-	if _, err := os.Stat(artifactPath); err != nil {
-		t.Fatalf("expected tracker torrent file: %v", err)
+	legacyBase := strings.TrimSuffix(strings.TrimPrefix(filepath.Base(torrentPath), "[ptp]."), ".torrent")
+	legacyPath := filepath.Join(filepath.Dir(torrentPath), legacyBase+".ptp.torrent")
+	if _, err := os.Stat(legacyPath); !errors.Is(err, os.ErrNotExist) {
+		t.Fatalf("expected no duplicate PTP torrent at %q, got %v", legacyPath, err)
```
[the remark this answers is not in the document]
Comment (Audionut) [reply internal/trackers/impl/standalone/ptp/definition_test.go]: @coderabbitai Fixed in e090e22c1: the test independently checks [ptp].<source-base>.torrent and <source-base>.ptp.torrent. PTP's layout base comes from SourcePath/Release, not ReleaseName. Focused race test passes.
